homeassistant/components/zwave/__init__.py

```diff
@@ -240,27 +260,31 @@ def get_config_value(node, value_index, tries=5):
     from openzwave.network import ZWaveNetwork
     from openzwave.group import ZWaveGroup
 
+    config = {}
+    if DATA_ZWAVE_CONFIG in hass.data:
+        config = hass.data[DATA_ZWAVE_CONFIG]
+
     # Load configuration
-    use_debug = config[DOMAIN].get(CONF_DEBUG)
-    autoheal = config[DOMAIN].get(CONF_AUTOHEAL)
+    use_debug = config.get(CONF_DEBUG, DEFAULT_DEBUG)
+    autoheal = config.get(CONF_AUTOHEAL,
+                          DEFAULT_CONF_AUTOHEAL)
     device_config = EntityValues(
-        config[DOMAIN][CONF_DEVICE_CONFIG],
-        config[DOMAIN][CONF_DEVICE_CONFIG_DOMAIN],
-        config[DOMAIN][CONF_DEVICE_CONFIG_GLOB])
+        config.get(CONF_DEVICE_CONFIG),
+        config.get(CONF_DEVICE_CONFIG_DOMAIN),
+        config.get(CONF_DEVICE_CONFIG_GLOB))
 
     # Setup options
     options = ZWaveOption(
-        config[DOMAIN].get(CONF_USB_STICK_PATH),
+        config_entry.data[CONF_USB_STICK_PATH],
         user_path=hass.config.config_dir,
-        config_path=config[DOMAIN].get(CONF_CONFIG_PATH))
+        config_path=config.get(CONF_CONFIG_PATH))
 
     options.set_console_output(use_debug)
 
-    if CONF_NETWORK_KEY in config[DOMAIN]:
-        options.addOption("NetworkKey", config[DOMAIN][CONF_NETWORK_KEY])
+    if CONF_NETWORK_KEY in config_entry.data:
+        options.addOption("NetworkKey", config_entry.data[CONF_NETWORK_KEY])
 
     options.lock()
```

Comment:
Is this blocking or I/O?

Reply:
Looks like it is doing I/O 

https://github.com/OpenZWave/open-zwave/blob/master/cpp/src/Options.cpp#L357

We should run this in the executor.

```python
await hass.async_add_executor_job(options.lock)
```